tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs

```diff
@@ -0,0 +1,343 @@
+            }
+        }
+
+        internal static DebuggerManager Instance => _lazyInstance.Value;
+
+        internal DebuggerSettings DebuggerSettings { get; private set; }
+
+        internal ExceptionReplaySettings ExceptionReplaySettings { get; }
+
+        internal DynamicInstrumentation? DynamicInstrumentation { get; private set; }
+
+        internal SpanCodeOrigin.SpanCodeOrigin? CodeOrigin { get; private set; }
+
+        internal IDebuggerUploader? SymbolsUploader { get; private set; }
+
+        internal ExceptionDebugging? ExceptionReplay { get; private set; }
+
+        internal string ServiceName => DynamicInstrumentationHelper.ServiceName;
+
+        // /!\ This method is called by reflection in the Samples.SampleHelpers
+        // If you remove it then you need to provide an alternative way to wait for the discovery service
+        private static async Task<bool> WaitForDiscoveryService(IDiscoveryService discoveryService)
+        {
+            var tc = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            // Stop waiting if we're shutting down
+            LifetimeManager.Instance.AddShutdownTask(_ => tc.TrySetResult(false));
+
+            discoveryService.SubscribeToChanges(Callback);
+            return await tc.Task.ConfigureAwait(false);
+
+            void Callback(AgentConfiguration x)
+            {
+                tc.TrySetResult(true);
+                discoveryService.RemoveSubscription(Callback);
+            }
+        }
+
+        private void SetGeneralConfig(DebuggerSettings settings)
+        {
+            DebuggerSnapshotSerializer.SetConfig(settings);
+            Redaction.Instance.SetConfig(settings.RedactedIdentifiers, settings.RedactedExcludedIdentifiers, settings.RedactedTypes);
+        }
+
+        private async Task InitializeSymbolUploader()
+        {
+            try
+            {
+                var tracerManager = TracerManager.Instance;
+                SymbolsUploader = DebuggerFactory.CreateSymbolsUploader(tracerManager.DiscoveryService, RcmSubscriptionManager.Instance, tracerManager.Settings, Instance.ServiceName, Instance.DebuggerSettings, tracerManager.GitMetadataTagsProvider);
+                // it will do nothing if it is an instance of NoOpSymbolUploader
+                await SymbolsUploader.StartFlushingAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error initializing Symbol Database.");
+            }
+        }
+
+        private void SetCodeOriginState()
+        {
+            try
+            {
+                var enabled = DebuggerSettings.CodeOriginForSpansEnabled.HasValue && DebuggerSettings.CodeOriginForSpansEnabled.Value;
+                var disabled = DebuggerSettings.CodeOriginForSpansEnabled.HasValue && !DebuggerSettings.CodeOriginForSpansEnabled.Value;
```

Comment:
Nit: I would name them `manuallyEnabled` & `manuallyDisabled`

Reply:
Fixed in 8696b2fbcb8573790b216971339a8bb41e40ff32